plaidml/bridge/openvino/ops/non_max_suppression.cpp

```diff
@@ -0,0 +1,283 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset5.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;  // NOLINT[build/namespaces]
+using ngraph::opset5::NonMaxSuppression;
+
+namespace PlaidMLPlugin {
+
+class nms {
+ public:
+  explicit nms(edsl::Tensor BOXES, edsl::Tensor SCORES, int32_t max_output_boxes_per_class, edsl::Tensor IOU_THRESHOLD,
+               edsl::Tensor SCORE_THRESHOLD, edsl::Tensor SOFT_NMS_SIGMA)
+      : BOXES_(BOXES),
+        SCORES_(SCORES),
+        max_output_boxes_per_class_(max_output_boxes_per_class),
+        IOU_THRESHOLD_(IOU_THRESHOLD),
+        SCORE_THRESHOLD_(SCORE_THRESHOLD),
+        SOFT_NMS_SIGMA_(SOFT_NMS_SIGMA),
+        center_point_box_(false),
+        sort_result_descending_(false),
+        box_input_type_(DType::FLOAT32),
+        box_output_type_(DType::INT32),
+        thres_type_(DType::FLOAT32) {}
+
+  nms& center_point_box(bool center_point_box) {
+    center_point_box_ = center_point_box;
+    return *this;
+  }
+
+  nms& sort_result_descending(bool sort_result_descending) {
+    sort_result_descending_ = sort_result_descending;
+    return *this;
+  }
+
+  nms& box_input_type(DType box_input_type) {
+    box_input_type_ = box_input_type;
+    return *this;
+  }
+
+  nms& box_output_type(DType box_output_type) {
+    box_output_type_ = box_output_type;
+    return *this;
+  }
+
+  nms& thres_type(DType thres_type) {
+    thres_type_ = thres_type_;
+    return *this;
+  }
+
+  std::vector<edsl::Tensor> build();
+
+ private:
+  edsl::Tensor BOXES_;
+  edsl::Tensor SCORES_;
+  int32_t max_output_boxes_per_class_;
+  edsl::Tensor IOU_THRESHOLD_;
+  edsl::Tensor SCORE_THRESHOLD_;
+  edsl::Tensor SOFT_NMS_SIGMA_;
+  bool center_point_box_;
+  bool sort_result_descending_;
+  DType box_input_type_;
+  DType box_output_type_;
+  DType thres_type_;
+};
+
+std::vector<edsl::Tensor> nms::build() {
+  std::vector<int64_t> boxes_shape = BOXES_.compute_shape().sizes();
+  std::vector<int64_t> scores_shape = SCORES_.compute_shape().sizes();
+  int num_batches = boxes_shape[0];
+  int num_boxes = boxes_shape[1];
+  int box_size = 4;
+  int num_classes = scores_shape[1];
+
+  edsl::Tensor PIECE = edsl::cast(edsl::index({edsl::TensorDim(box_size)}, 0), box_input_type_);
+  edsl::Tensor ZERO = edsl::reshape(op::slice(PIECE).add_dims({0}), {edsl::TensorDim(1)});
```

Comment:
Could you please change the name `ONE` 'TWO' et al to something more descriptive ? At a first glance, I would assume it is a constant tensor 1 or 2.

Reply:
They mean a memref with the constant value, the reason to do this way is  the second parameter of gather must be a memref, or it will crash.    So how about ONE_MEMREF ? or ONE_INPUTTYPE?

Reply:
I see. I misunderstood this section, I thought is was a tensor passed from out side. 
I think we can use something like the following,
```
ZERO=edsl::cast(edsl::index({edsl::TensorDim(1)}, 0), box_input_type_);
ONE=ZERO+1;
TWO=ZERO+2;
...
```

Reply:
Yes, this is a good idea

Reply:
Done